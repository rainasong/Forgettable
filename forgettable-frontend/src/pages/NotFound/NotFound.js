import React from 'react';
import classes from './NotFound.module.css';
import LightLogo from '../../assets/logos/logo-black.svg';
import DarkLogo from '../../assets/logos/logo-white.svg';
import IconButton from '../../components/IconButton/IconButton';
import {useNavigate} from 'react-router-dom';
/*
 * This is the 404 Not Found page. All invalid URLs will
 * redirect to here.
 *
 * Author: Mercury Lin (lin8231)
 */
const NotFound = (props) => {
  const navigate = useNavigate();

  const theme = localStorage.getItem('theme');

  return (
    <div className={classes.NotFound}>
      <div className={classes.ContentContainer}>
        <h1>
          {'Oops! We can\'t seem to find'}
          <br/>
          {'the page you\'re looking for :('}
        </h1>
        <p>Error 404 - Page Not Found</p>
        <IconButton
          btnText="Go back to homepage"
          height="60px"
          onClick={() => navigate('/')}
        />
      </div>
      <img
        src={theme == 'dark' ? DarkLogo : LightLogo}
        alt="Forgettable Logo"
        className={classes.Logo}
      />
    </div>
  );
};

export default NotFound;
