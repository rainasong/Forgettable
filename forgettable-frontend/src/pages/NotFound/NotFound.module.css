.NotFound {
    height: 100vh;
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    overflow: hidden;
}

.ContentContainer {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: -5%;
}

.ContentContainer > h1 {

}

.ContentContainer > p {
    margin-bottom: 50px;
    margin-top: 0;
}

.Logo {
    position: absolute;
    width:250px;
    bottom: 20%;
}