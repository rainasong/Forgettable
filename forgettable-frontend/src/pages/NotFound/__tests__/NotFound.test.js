
import ShallowRenderer from 'react-test-renderer/shallow';
import NotFound from '../NotFound';
import React from 'react';
import {BrowserRouter} from 'react-router-dom';

it('renders NotFound UI to with the correct hierarchy', () => {
  const renderer = new ShallowRenderer();
  renderer.render(
      <BrowserRouter>
        <NotFound/>
      </BrowserRouter>,
  );
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});

