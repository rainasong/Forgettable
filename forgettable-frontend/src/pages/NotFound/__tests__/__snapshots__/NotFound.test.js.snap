// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NotFound UI to with the correct hierarchy 1`] = `
<Router
  location={
    Object {
      "hash": "",
      "key": "default",
      "pathname": "/",
      "search": "",
      "state": null,
    }
  }
  navigationType="POP"
  navigator={
    Object {
      "action": "POP",
      "back": [Function],
      "block": [Function],
      "createHref": [Function],
      "forward": [Function],
      "go": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "default",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <NotFound />
</Router>
`;
